docs/data-guidelines/api.md

```diff
@@ -2,6 +2,49 @@
 
 This file contains guidelines that are specific to the web API features (`api/`).
 
+
+## Adding features and subfeatures
+
+A json file should be created for every _top level feature_ that has a released public implementation on one of the supported set of browsers (and not before).
+For Web APIs the top level features are individual Web API interfaces and the children of namespace objects like `Intl`.
+
+A possible structure for a Web API is shown below. 
+
+```
+api                           // category - always api for Web APIs
+  interface                   // top level feature
+    properties               // feature: required
+    methods                  // feature: required
+       parameters             //   subfeature: compatibility is different
+         parameter options    //   subfeature: if compatibility is different
+    support in workers     // subfeature: if compatibility is different (might also just be on specific methods)
+```
+
+The interface must include subnodes for _all_ its properties and methods (that have an implementation in at least one of the supported browsers), whether or not they were added at the same time as the parent interface, and hence share the same compatibility information.
+
+All other code and behavioral aspects of the interface are given nested nodes **only** if their compatibility is _different_ to that of their parent feature.
+This applies to method arguments, method argument options, and behaviour changes like the addition of support for workers, or the type of return value.
+
+> **Note:** Items that must be given nodes are referred to as "features", while those that are only created when there is a compatibility difference to the parent are called _subfeatures_.
```

Comment:
@foolip See https://github.com/mdn/browser-compat-data/pull/17352/files#r972516073

I would be happy with this if we used feature and subfeature sanely. But we don't. So define it properly - otherwise node and sub node are better.

If I sound frustrated, it is because I am. So much time wasted on this. IMO if we just said "feature" is the top level item and subfeature is any nested item at any nesting level this would be fine. But you'll have to get @queengooborg to agree - I couldn't

Reply:
Huh?  But that's exactly the definition of "subfeature" I was hoping for...

Reply:
@queengooborg I really hope I haven't been in "violent agreement" with you for most of this discussion :-0.